gtsam/linear/NoiseModel.cpp

```diff
@@ -135,14 +135,25 @@ bool Gaussian::equals(const Base& expected, double tol) const {
   const Gaussian* p = dynamic_cast<const Gaussian*> (&expected);
   if (p == nullptr) return false;
   if (typeid(*this) != typeid(*p)) return false;
-  //if (!sqrt_information_) return true; // ALEX todo;
+  //TODO(Alex);
+  //if (!sqrt_information_) return true;
   return equal_with_abs_tol(R(), p->R(), sqrt(tol));
 }
 
+/* ************************************************************************* */
+Matrix Gaussian::covariance() const {
+  // Uses a fast version of `covariance = information().inverse();`
+  Matrix R = this->R();
```

Comment:
You are making an unnecessary copy here. `const Matrix&`

Reply:
Nice catch!